PlatformIO build fails due to missing gmock dependency in StreamGMock

## Description

When using ArduinoMock as a PlatformIO `lib_deps` dependency, the build fails because PlatformIO compiles **all** source files in `src/`, including `stream/StreamGMock.cpp`, which requires `gmock/gmock.h`. Since Google Mock is not bundled or declared as a dependency, the build fails with:

```
.pio/libdeps/native/ArduinoMock/src/stream/StreamGMock.cpp:1:
stream/StreamGMock.h:3:10: fatal error: 'gmock/gmock.h' file not found
    3 | #include <gmock/gmock.h>
```

The CMake build already handles this correctly via `ARDUINOMOCK_USE_GMOCK` (OFF by default), which excludes `StreamGMock` files when gmock is not needed (CMakeLists.txt lines 26-28). But `library.json` has no equivalent, so PlatformIO users always hit this error.

## Proposed Fix

Update `library.json` to exclude `StreamGMock.cpp` from the default build using `srcFilter`, mirroring the CMake default:

```json
{
  "name": "ArduinoMock",
  "version": "0.1.1-rc.1",
  "description": "Arduino native mocks for host-based testing",
  "platforms": ["native"],
  "build": {
    "srcFilter": ["+<*>", "-<stream/StreamGMock.cpp>"]
  }
}
```

This has been tested and confirmed working with `pio test -e native`.

## Reproduction

In `platformio.ini`:
```ini
[env:native]
platform = native
lib_deps =
  https://github.com/ByteNana/ArduinoMock#master
```

Run `pio test -e native` → build fails on `gmock/gmock.h` not found.

--- a/library.json
+++ b/library.json
@@ -1,0 +1,9 @@
+{
+  "name": "ArduinoMock",
+  "version": "0.2.0-rc.1",
+  "description": "Arduino native mocks for host-based testing",
+  "platforms": ["native"],
+  "build": {
+    "srcFilter": ["+<*>", "-<stream/StreamGMock.cpp>"]
+  }
+}

